nmproxy/metrics/metrics_helper.go

```diff
@@ -1,36 +1,24 @@
 package metrics
 
 import (
-	"fmt"
 	"time"
 
-	"github.com/go-ping/ping"
-	"github.com/gravitl/netmaker/logger"
+	"github.com/gravitl/netclient/nmproxy/config"
+	"github.com/gravitl/netclient/nmproxy/wg"
 )
 
 const MetricCollectionInterval = time.Second * 25
 
-// PeerConnectionStatus - get peer connection status by pinging
-func PeerConnectionStatus(address string) (connected bool) {
-	fmt.Println("PINGER ADDR: ", address)
-	pinger, err := ping.NewPinger(address)
+// PeerConnectionStatus - get peer connection status from wireguard interface
+func PeerConnectionStatus(peerPublicKey string) bool {
+	ifacePeers, err := wg.GetPeers(config.GetCfg().GetIface().Name)
 	if err != nil {
-		logger.Log(0, "could not initiliaze ping peer address", address, err.Error())
-		connected = false
-	} else {
-		pinger.Timeout = time.Second * 2
-		err = pinger.Run()
-		if err != nil {
-			logger.Log(0, "failed to ping on peer address", address, err.Error())
-			return false
-		} else {
-			pingStats := pinger.Statistics()
-			if pingStats.PacketsRecv > 0 {
-				connected = true
-				return
-			}
+		return false
+	}
+	for _, peer := range ifacePeers {
+		if peer.PublicKey.String() == peerPublicKey {
+			return peer.LastHandshakeTime.After(time.Now().Add(-3*time.Minute)) && peer.ReceiveBytes+peer.TransmitBytes > 0
```

Comment:
`time.Now().Sub()` method can be used

Reply:
time.Sub() returns time.Duration which additionally needs to be casted to time.Time

but time.Add returns time.Time